README.md

```diff
@@ -269,18 +271,34 @@ To prevent this put double dash `--` before APP. It will make pipx to forward th
 
 Re-running the same app is quick because pipx caches Virtual Environments on a per-app basis. The caches only last a few days, and when they expire, pipx will again use the latest version of the package. This way you can be sure you're always running a new version of the package without having to manually upgrade.
 
-If the app name does not match that package name, you can use the `--spec` argument to specify the package to install and app to run separately:
+### Package with multiple apps, or the app name doesn't match the package name
+
+If the app name does not match the package name, you can use the `--spec` argument to specify the `PACKAGE` name, and provide the `APP` to run separately:
 
 ```
 pipx run --spec PACKAGE APP
 ```
 
-You can also specify specific versions, version ranges, or extras:
+For example:
+
+```
+pipx run --spec esptool esptool.py
+pipx run --spec esptool espsecure.py
```

Comment:
Have you tried running these apps without the filetype ending?

Reply:
Yes, and it doesn't work. The project [esptool](https://github.com/espressif/esptool) provides executables with `.py` ending:

```
$ pipx run --spec esptool esptool
'esptool' executable script not found in package 'esptool'.
Available executable scripts:
    esp_rfc2217_server.py - usage: 'pipx run --spec esptool esp_rfc2217_server.py [arguments?]'
    espefuse.py - usage: 'pipx run --spec esptool espefuse.py [arguments?]'
    espsecure.py - usage: 'pipx run --spec esptool espsecure.py [arguments?]'
    esptool.py - usage: 'pipx run --spec esptool esptool.py [arguments?]'
```

I suppose this is different from package to package. There are projects that provide executables without the `.py` ending:

```
$ pipx run --spec pymodbus[repl] pymodbus.simulator 
2023-06-26 09:35:23,311 INFO  logging:96 Start simulator
2023-06-26 09:35:23,430 INFO  logging:96 Modbus server started
Executing <Task pending name='Task-1' coro=<run_main() running at /home/william/.local/pipx/.cache/8d7f1cb8ac876e4/lib/python3.11/site-packages/pymodbus/server/simulator/main.py:116> wait_for=<_GatheringFuture pending cb=[Task.task_wakeup()] created at /usr/lib/python3.11/asyncio/tasks.py:699> cb=[_run_until_complete_cb() at /usr/lib/python3.11/asyncio/base_events.py:180] created at /usr/lib/python3.11/asyncio/runners.py:100> took 0.119 seconds
2023-06-26 09:35:23,431 INFO  logging:96 HTTP server started
2023-06-26 09:35:23,431 INFO  logging:96 Server(TCP) listening.
```

Now that I think about it, it seems to be a good example to add into the README. The package requirement has a `[repl]` suffix, which pulls in dependencies for a REPL, in this instance.

Reply:
Can we add some more text in the document to explain this? i.e. list out what scripts `esptool` provides so people not familiar with this particular package understand the exact situation.

Reply:
Alright, I've added more detailed explanation in my latest commit.